hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/security/token/delegation/DelegationTokenSecretManager.java

```diff
@@ -401,7 +402,10 @@ protected void logExpireToken(final DelegationTokenIdentifier dtId)
       // closes the edit log files. Doing this inside the
       // fsn lock will prevent being interrupted when stopping
       // the secret manager.
-      namesystem.readLockInterruptibly();
+      // TODO: delegation token is a very independent system, so
+      // it's proper to use an seperated r/w lock instead of fs lock
+      // for getting/renewing/expiring/canceling token or updating master key.
```

Comment:
`logUpdateMasterKey` and `logExpireDelegationToken` need to write edit log, so HDFS-13112 add `hasReadLock` for these methods. 

So I think this FSLock is needed before we understand why `rollEdits` is not thread-safe as HDFS-13112 said.

Reply:
@yuanboliu @ZanderXu modify the comments or keep it there ?